include/revng/ABI/Definition.h

```diff
@@ -466,10 +466,12 @@ class Definition : public generated::Definition {
 
     for (auto Register : GPRs)
       if (Lookup.contains(Register))
-        Result.emplace_back(Register);
+        if (!llvm::is_contained(Result, Register))
+          Result.emplace_back(Register);
     for (auto Register : VectorArgumentRegisters())
       if (Lookup.contains(Register))
-        Result.emplace_back(Register);
+        if (!llvm::is_contained(Result, Register))
+          Result.emplace_back(Register);
```

Comment:
The message commit is quite vague, what faulty situation are we trying to prevent?
Can you make an example?

Reply:
The case where the same register is mentioned more than once. I found out about this limitation when trying to encode `gcc's mips` ABI before adding `FloatsAlwaysUseGenericRegisters` flag.

But also, consider, that this is a register-_sorting_ method. I don't see a reason for it to complain about duplication. Let ABI definition specify whichever registers it wants - all this cares about is their _order_.

Reply:
I'd prefer to assert and force callers to dedup things. If have 5 registers and I ask someone to sort them and I get back 4, I'd like an abort.

Reply:
Actually, wait, yeah. That's my bad on the implementation (I just sticked "if contained" thing in without thinking). This needs a bit of a rewrite. The intention is to deduplicate the { `ReturnValueLocationRegister` + `GeneralPurposeArgumentRegisters` + `VectorArgumentRegisters` } set, not the callee's input.

Thanks for catching this!

Reply:
Fixed.

I also took this opportunity to modernize this code.
Please review the new `ABI: redesign register sorting from the ground up` commit.